### ISSUE TITLE
panic serving, interface conversion: interface {} is nil, not string

### ISSUE THREAD

**Issue by kavehmz** [dgraph-io/dgraph#3201]
I am using `v1.0.13`.

For some of my queries I am getting the following panic:

```
2019/03/25 09:43:14 http: panic serving 127.0.0.1:55570: interface conversion: interface {} is nil, not string
goroutine 2758 [running]:
net/http.(*conn).serve.func1(0xc03bfcb860)
	/usr/local/go/src/net/http/server.go:1746 +0xd0
panic(0x134a840, 0xc02fb67b60)
	/usr/local/go/src/runtime/panic.go:513 +0x1b9
github.com/dgraph-io/dgraph/types.Marshal(0x0, 0x0, 0x0, 0xc010f852e0, 0xc02fb67b00, 0x1ef83b0)
	/ext-go/1/src/github.com/dgraph-io/dgraph/types/conversion.go:349 +0x108a
github.com/dgraph-io/dgraph/query.(*SubGraph).replaceVarInFunc(0xc022ef4b00, 0xc028f6de90, 0xc017372c46)
	/ext-go/1/src/github.com/dgraph-io/dgraph/query/query.go:1648 +0x3e9
github.com/dgraph-io/dgraph/query.(*SubGraph).fillVars(0xc022ef4b00, 0xc028f6de90, 0x0, 0x0)
	/ext-go/1/src/github.com/dgraph-io/dgraph/query/query.go:1619 +0x249
github.com/dgraph-io/dgraph/query.(*SubGraph).recursiveFillVars(0xc022ef4b00, 0xc028f6de90, 0x0, 0x0)
	/ext-go/1/src/github.com/dgraph-io/dgraph/query/query.go:1559 +0x39
github.com/dgraph-io/dgraph/query.(*SubGraph).recursiveFillVars(0xc022ef42c0, 0xc028f6de90, 0x0, 0x0)
	/ext-go/1/src/github.com/dgraph-io/dgraph/query/query.go:1570 +0x119
github.com/dgraph-io/dgraph/query.(*SubGraph).recursiveFillVars(0xc027947b80, 0xc028f6de90, 0x0, 0x0)
	/ext-go/1/src/github.com/dgraph-io/dgraph/query/query.go:1570 +0x119
github.com/dgraph-io/dgraph/query.(*SubGraph).recursiveFillVars(0xc027946b00, 0xc028f6de90, 0x0, 0x0)
	/ext-go/1/src/github.com/dgraph-io/dgraph/query/query.go:1570 +0x119
github.com/dgraph-io/dgraph/query.(*SubGraph).recursiveFillVars(0xc027946840, 0xc028f6de90, 0x0, 0x0)
	/ext-go/1/src/github.com/dgraph-io/dgraph/query/query.go:1570 +0x119
github.com/dgraph-io/dgraph/query.(*SubGraph).recursiveFillVars(0xc0279462c0, 0xc028f6de90, 0xc028f68c00, 0x1)
	/ext-go/1/src/github.com/dgraph-io/dgraph/query/query.go:1564 +0xa6
github.com/dgraph-io/dgraph/query.(*QueryRequest).ProcessQuery(0xc010f85b58, 0x15acba0, 0xc04bfe6f00, 0x1, 0x1)
	/ext-go/1/src/github.com/dgraph-io/dgraph/query/query.go:2526 +0x61d
github.com/dgraph-io/dgraph/query.(*QueryRequest).Process(0xc010f85b58, 0x15acba0, 0xc04bfe6f00, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/ext-go/1/src/github.com/dgraph-io/dgraph/query/query.go:2620 +0x5e
github.com/dgraph-io/dgraph/edgraph.(*Server).Query(0xc00eeaec20, 0x15acba0, 0xc04bfe6f00, 0xc0322a17c0, 0xc03b7d8060, 0x0, 0x0)
	/ext-go/1/src/github.com/dgraph-io/dgraph/edgraph/server.go:506 +0x4e4
github.com/dgraph-io/dgraph/dgraph/cmd/alpha.queryHandler(0x15abb60, 0xc0004c47e0, 0xc00c9d1400)
	/ext-go/1/src/github.com/dgraph-io/dgraph/dgraph/cmd/alpha/http.go:175 +0x271
net/http.HandlerFunc.ServeHTTP(0x14c2f88, 0x15abb60, 0xc0004c47e0, 0xc00c9d1400)
	/usr/local/go/src/net/http/server.go:1964 +0x44
net/http.(*ServeMux).ServeHTTP(0x1fa2120, 0x15abb60, 0xc0004c47e0, 0xc00c9d1400)
	/usr/local/go/src/net/http/server.go:2361 +0x127
net/http.serverHandler.ServeHTTP(0xc00057f6c0, 0x15abb60, 0xc0004c47e0, 0xc00c9d1400)
	/usr/local/go/src/net/http/server.go:2741 +0xab
net/http.(*conn).serve(0xc03bfcb860, 0x15acae0, 0xc04371cbc0)
	/usr/local/go/src/net/http/server.go:1847 +0x646
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:2851 +0x2f5
```

I assume it might have been fixed in `v1.0.14-rc1` but if not it needs review.

**Comment by kavehmz** [comment]
I switched back to `v1.0.11` and not I see the correct errors instead of panic for the same query.

```
            "code": "ErrorInvalidRequest",
            "message": ": No value found for value variable \"DESC\""
```

the issue, if not fixed already, should be easy to reproduce and fix.

**Comment by srfrog** [comment]
Thanks, for reporting this. Apparently some changes didn't make to the .13 release which caused this panic. The master version has the updated function.

**Comment by martinmr** [comment]
It looks like the fix went out in the .13 release.

The commit in question ecad42e1a2b0d752c654add5e3bb94c76d7fecb3 from Feb 21th.

@srfrog Can you verify if this panic is indeed caused by the same bug the fix was supposed to remove? Maybe this is a different bug.

**Comment by srfrog** [comment]
The fix are in commits https://github.com/dgraph-io/dgraph/pull/2995/commits/4501a71fafa29cf8fc723465a2bc39c48af2fc63 and https://github.com/dgraph-io/dgraph/pull/2995/commits/aeb058dbf89445baebe70da2bed70646ec60f0df which are already merged to master.

**Comment by martinmr** [comment]
but the should also make it into the 1.0 branch, shouldn't they?

**Comment by srfrog** [comment]
Changes merged to release/v1.0 and they will be out in next release. Closing this.

**Comment by erhlee-bird** [comment]
I think I'm still seeing this in v1.0.14

schema:
```
<number>: int @index(int) .
```

query:
```
{
  var(func: has(number, orderdesc: number, first: 1) 
  {
    highest_number as number
  }

  all(func: eq(number, val(highest_number))) 
  { 
    uid 
  }
}
```

responses:
v1.0.11 - grpc_message 'No value found for value variable "highest_number"' and no panic
v1.0.13 - grpc_message "Socket closed" and the panic described in the title
v1.0.14 - same as v1.0.13

**Comment by erhlee-bird** [comment]
@srfrog can we reopen this issue?

I've been able to reliably reproduce this from v1.0.12 - v1.0.15-rc3 but not v1.0.11.

This happens when I spin up a fresh Dgraph cluster (1 zero, 1 alpha as well as 1 zeros, 3 alphas)
and make the query without any data present in the database.

**Comment by erhlee-bird** [comment]
I've confirmed that this is fixed in v1.0.16

**Comment by danielmai** [comment]
Thanks for confirming, @erhlee-bird. Closing this issue since it's fixed in v1.0.16 by #3554.
